### PR TITLE
Components fixes

### PR DESCRIPTION
**Pull request by igor-davidyuk**

### CODE REVIEW

File: docs/source/openfl/components.rst [hunk 1]
```diff
@@ -31,13 +31,18 @@ These components are terminated when the experiment is finished.
...
+The *Aggregator* is framework-agnostic, as it operates tensors in OpenFL inner representation, 
+while the *Collaborator* can use deep learning frameworks as computational backend, such as `TensorFlow* <https://www.tensorflow.org/>`_ or `PyTorch* <https://pytorch.org/>`_.
 
 
 Aggregator
 ----------
 
-The Aggregator is a short-lived entity, which means that its lifespan is limited by the experiment execution time. It orchestrates Collaborators according to the FL plan and performs model aggregation at the end of each round. 
+The Aggregator is a short-lived entity, which means that its lifespan is limited by the experiment execution time. 
+It orchestrates Collaborators according to the FL plan, performs model aggregation at the end of each round,
```
Comment (aleksandr-mokrov) [inline docs/source/openfl/components.rst:42]: ... each round and acts as a parameter server for collaborators.

File: docs/source/openfl/components.rst [hunk 2]
```diff
@@ -31,13 +31,18 @@ These components are terminated when the experiment is finished.
...
 
 Aggregator
 ----------
 
-The Aggregator is a short-lived entity, which means that its lifespan is limited by the experiment execution time. It orchestrates Collaborators according to the FL plan and performs model aggregation at the end of each round. 
+The Aggregator is a short-lived entity, which means that its lifespan is limited by the experiment execution time. 
+It orchestrates Collaborators according to the FL plan, performs model aggregation at the end of each round, 
+and acts as parametr server for collaborators. 
+
+Model weights aggregation logic may be customized via :ref:`plugin mechanism <overriding_agg_fn>`.
```
Comment (aleksandr-mokrov) [inline docs/source/openfl/components.rst]:
```suggestion
Model weight aggregation logic may be customized via :ref:`plugin mechanism <overriding_agg_fn>`.
```

File: docs/source/openfl/components.rst [hunk 3]
```diff
@@ -51,41 +56,48 @@ The Collaborator is a short-lived entity that manages training the model on loca
...
 
-Every Collaborator is a unique service. The model is loaded with a local *shard descriptor* to perform tasks included in an FL experiment. At the end of the training task, weights are extracted and sent to the central node and aggregated.
+Every Collaborator is a unique service. The data loader is loaded with a local *shard descriptor* to perform tasks 
+included in an FL experiment. At the end of the training task, weights are extracted and sent to the central node 
+and aggregated.
 
-Converting tensor objects is handled by :ref:`framework adapter <framework_adapter>` plugins. Included in |productName| are framework adapters for PyTorch and TensorFlow 2.x. These framework adapters are extensible. You can contribute new framework adapters for deep learning frameworks you would like see supported in |productName|. 
+Converting tensor objects is handled by :ref:`framework adapter <framework_adapter>` plugins. 
+Included in |productName| are framework adapters for PyTorch and TensorFlow 2.x. 
+The list of framework adapters is extensible. User can contribute new framework adapters for deep learning frameworks
```
Comment (aleksandr-mokrov) [inline docs/source/openfl/components.rst:69]: `User can contribute ... they would like see ...` or  `You can contribute ... you would like see ...`

File: docs/source/openfl/components.rst [hunk 4]
```diff
@@ -51,41 +56,48 @@ The Collaborator is a short-lived entity that manages training the model on loca
...
 
-The Director is a long-lived entity and is the central node of the federation. It runs two services: 
-
-    - frontend Director client
-    - interactive Python API
+The Director is a long-lived entity and is the central node of the federation. It accepts connections from: 
 
-The Director supports concurrent frontend connections. While the Director may take in several experiments (with the same data interface), the experiments are executed in series.
+    - Frontend clients (data scientists using :ref:`interactive_api`)
+    - Envoys, if their Shar Descriptors are complient to the same data interface
```
Comment (aleksandr-mokrov) [inline docs/source/openfl/components.rst]: Shard Descriptors

File: docs/source/openfl/components.rst [hunk 5]
```diff
@@ -95,8 +107,12 @@ Envoy
 
 The Envoy is a long-lived entity that runs on collaborator nodes connected to the Director. 
 
-Every Envoy is matched to one `shard descriptor <https://github.com/intel/openfl/blob/develop/openfl/interface/interactive_api/shard_descriptor.py>`_ in order to run. When the Director starts an experiment, the Envoy accepts the experiment workspace, prepares the environment, and starts a Collaborator.
+Every Envoy is matched to one `shard descriptor <https://github.com/intel/openfl/blob/develop/openfl/interface/interactive_api/shard_descriptor.py>`_ 
+in order to run. When the Director starts an experiment, the Envoy accepts the experiment workspace, 
+prepares the environment, and starts a Collaborator.
 
+Envoy is also responsible for sending heartbeat messages to the Director, this messages may include information
```
Comment (aleksandr-mokrov) [inline docs/source/openfl/components.rst]: The Envoy ... to the Director. This messages ...

**Review by aleksandr-mokrov — CHANGES_REQUESTED** [review]

File: docs/source/openfl/components.rst [hunk 6]
```diff
@@ -95,8 +107,12 @@ Envoy
 
 The Envoy is a long-lived entity that runs on collaborator nodes connected to the Director. 
 
-Every Envoy is matched to one `shard descriptor <https://github.com/intel/openfl/blob/develop/openfl/interface/interactive_api/shard_descriptor.py>`_ in order to run. When the Director starts an experiment, the Envoy accepts the experiment workspace, prepares the environment, and starts a Collaborator.
+Every Envoy is matched to one `shard descriptor <https://github.com/intel/openfl/blob/develop/openfl/interface/interactive_api/shard_descriptor.py>`_ 
+in order to run. When the Director starts an experiment, the Envoy accepts the experiment workspace, 
+prepares the environment, and starts a Collaborator.
 
+Envoy is also responsible for sending heartbeat messages to the Director, this messages may include information 
+regarding the collaborator machine resources utilization.
```
Comment (aleksandr-mokrov) [inline docs/source/openfl/components.rst]: probably `regarding collaborator machine resource utilization.`?

File: docs/source/openfl/components.rst (hunk 3 above)
Comment (igor-davidyuk) [reply docs/source/openfl/components.rst:69]: fixed thanks

File: docs/source/openfl/components.rst (hunk 4 above)
Comment (igor-davidyuk) [reply docs/source/openfl/components.rst]: done

File: docs/source/openfl/components.rst [hunk 7]
```diff
@@ -31,13 +31,18 @@ These components are terminated when the experiment is finished.
...
+while the *Collaborator* can use deep learning frameworks as computational backend, such as `TensorFlow* <https://www.tensorflow.org/>`_ or `PyTorch* <https://pytorch.org/>`_.
 
 
 Aggregator
 ----------
 
-The Aggregator is a short-lived entity, which means that its lifespan is limited by the experiment execution time. It orchestrates Collaborators according to the FL plan and performs model aggregation at the end of each round. 
+The Aggregator is a short-lived entity, which means that its lifespan is limited by the experiment execution time. 
+It orchestrates Collaborators according to the FL plan, performs model aggregation at the end of each round, 
+and acts as parametr server for collaborators.
```
Comment (igor-davidyuk) [inline docs/source/openfl/components.rst]:
```suggestion
and acts as a parameter server for collaborators. 
```

File: docs/source/openfl/components.rst [hunk 8]
```diff
@@ -95,8 +107,12 @@ Envoy
 
 The Envoy is a long-lived entity that runs on collaborator nodes connected to the Director. 
 
-Every Envoy is matched to one `shard descriptor <https://github.com/intel/openfl/blob/develop/openfl/interface/interactive_api/shard_descriptor.py>`_ in order to run. When the Director starts an experiment, the Envoy accepts the experiment workspace, prepares the environment, and starts a Collaborator.
+Every Envoy is matched to one `shard descriptor <https://github.com/intel/openfl/blob/develop/openfl/interface/interactive_api/shard_descriptor.py>`_ 
+in order to run. When the Director starts an experiment, the Envoy accepts the experiment workspace, 
+prepares the environment, and starts a Collaborator.
 
+Envoy is also responsible for sending heartbeat messages to the Director. These messages may also include information
```
Comment (aleksandr-mokrov) [inline docs/source/openfl/components.rst]:
```suggestion
The envoy is also responsible for sending heartbeat messages to the Director. These messages may also include information 
```

File: docs/source/openfl/components.rst [hunk 9]
```diff
@@ -95,8 +107,12 @@ Envoy
 
 The Envoy is a long-lived entity that runs on collaborator nodes connected to the Director. 
 
-Every Envoy is matched to one `shard descriptor <https://github.com/intel/openfl/blob/develop/openfl/interface/interactive_api/shard_descriptor.py>`_ in order to run. When the Director starts an experiment, the Envoy accepts the experiment workspace, prepares the environment, and starts a Collaborator.
+Every Envoy is matched to one `shard descriptor <https://github.com/intel/openfl/blob/develop/openfl/interface/interactive_api/shard_descriptor.py>`_ 
+in order to run. When the Director starts an experiment, the Envoy accepts the experiment workspace, 
+prepares the environment, and starts a Collaborator.
 
+Envoy is also responsible for sending heartbeat messages to the Director. These messages may also include information 
+regarding the collaborator machine resources utilization. Refer to `device_monitor_plugin`_ for details.
```
Comment (aleksandr-mokrov) [inline docs/source/openfl/components.rst]:
```suggestion
regarding collaborator machine resource utilization. Refer to `device_monitor_plugin`_ for details.
```

File: docs/source/openfl/components.rst [hunk 10]
```diff
@@ -51,41 +56,48 @@ The Collaborator is a short-lived entity that manages training the model on loca
...
     - exchanging model parameters with the Aggregator.
 
-The Collaborator is created by the :ref:`Envoy <openfl_ll_components_envoy>` when a new experiment is submitted. 
+The Collaborator is created by the :ref:`Envoy <openfl_ll_components_envoy>` when a new experiment is submitted 
+in the :ref:`Director-based workflow <director_workflow>`. The Collaborator should be started from CLI if a user follows the 
+:ref:`Aggregator-based workflow <running_the_federation_aggregator_based>`
 
-Every Collaborator is a unique service. The model is loaded with a local *shard descriptor* to perform tasks included in an FL experiment. At the end of the training task, weights are extracted and sent to the central node and aggregated.
+Every Collaborator is a unique service. The data loader is loaded with a local *shard descriptor* to perform tasks 
+included in an FL experiment. At the end of the training task, weights are extracted and sent to the central node
```
Comment (igor-davidyuk) [inline docs/source/openfl/components.rst]:
```suggestion
included in an FL experiment. At the end of the training task, weight tensors are extracted and sent to the central node 
```

**Review by aleksandr-mokrov — APPROVED** [review]

**Review by alexey-gruzdev — APPROVED** [review]

**Pull request merged**